### PR TITLE
[IMP] hr_attendance: Add overtime reporting

### PR DESCRIPTION
**Pull request by jugj-odoo**
task-3894091

### CODE REVIEW

File: addons/hr_attendance/views/hr_attendance_overtime_view.xml [hunk 1]
```diff
@@ -19,13 +19,64 @@
...
+            <p class="o_view_nocontent_empty_folder">
+                No overtime records found
+            </p><p>
+                The overtime records of your employees will be displayed here.
+            </p>
+        </field>
+
     </record>
+
+    <menuitem id="menu_hr_attendance_overtime_reporting" name="Extra Hours" parent="menu_hr_attendance_reporting" sequence="15" groups="hr_attendance.group_hr_attendance_officer" action="hr_attendance_overtime_action"/>
```
Comment (sofiagvaladze) [inline addons/hr_attendance/views/hr_attendance_overtime_view.xml:80]: It's my personal opinion, but I think it's better to put menuitems together.. so move this to addons/hr_attendance/views/hr_attendance_view.xml

**Review by sofiagvaladze — COMMENTED** [review]
Minor comment :raised_hands:

File: addons/hr_attendance/views/hr_attendance_overtime_view.xml [hunk 2]
```diff
@@ -19,13 +19,64 @@
             <search>
                 <field name="employee_id"/>
                 <field name="duration" filter_domain="[('duration', '>=', self)]"/>
+                <filter string="Last 3 Months" invisible="1" name="last_three_months" domain="[(
+                'date','&gt;=', (
+                    context_today() + datetime.timedelta(days=-90)
```
Comment (mepe-odoo) [inline addons/hr_attendance/views/hr_attendance_overtime_view.xml:24]: You can use relative delta to have a better precision :)
```suggestion
                    context_today() + relativedelta(months=-3)
```

**Review by mepe-odoo — APPROVED** [review]
LGTM just a small comment about timedelta :)

File: addons/hr_attendance/views/hr_attendance_view.xml [hunk 3]
```diff
@@ -289,7 +289,9 @@
 
     <menuitem id="menu_hr_attendance_kiosk_no_user_mode" name="Kiosk Mode" parent="menu_hr_attendance_root" sequence="10" groups="hr_attendance.group_hr_attendance_manager" action="open_kiosk_url"/>
 
-    <menuitem id="menu_hr_attendance_reporting" name="Reporting" parent="menu_hr_attendance_root" sequence="15" groups="hr_attendance.group_hr_attendance_officer" action="hr_attendance_reporting"/>
+    <menuitem id="menu_hr_attendance_reporting" name="Reporting" parent="menu_hr_attendance_root" sequence="15" groups="hr_attendance.group_hr_attendance_officer"/>
+
+    <menuitem id="menu_hr_attendance_attendance_reporting" name="Worked Hours" parent="menu_hr_attendance_reporting" sequence="15" groups="hr_attendance.group_hr_attendance_officer" action="hr_attendance_reporting"/>
```
Comment (Bertrand2) [inline addons/hr_attendance/views/hr_attendance_view.xml:294]: Line length :innocent: 
```suggestion
    <menuitem id="menu_hr_attendance_reporting" name="Reporting"
        parent="menu_hr_attendance_root" sequence="15"
        groups="hr_attendance.group_hr_attendance_officer"/>

    <menuitem id="menu_hr_attendance_attendance_reporting" name="Worked Hours"
        parent="menu_hr_attendance_reporting" sequence="15"
        groups="hr_attendance.group_hr_attendance_officer" action="hr_attendance_reporting"/>
```

File: addons/hr_attendance/views/hr_attendance_overtime_view.xml [hunk 4]
```diff
@@ -19,13 +19,64 @@
...
+            <p class="o_view_nocontent_empty_folder">
+                No overtime records found
+            </p><p>
+                The overtime records of your employees will be displayed here.
+            </p>
+        </field>
+
     </record>
+
+    <menuitem id="menu_hr_attendance_overtime_reporting" name="Extra Hours" parent="menu_hr_attendance_reporting" sequence="15" groups="hr_attendance.group_hr_attendance_officer" action="hr_attendance_overtime_action"/>
```
Comment (Bertrand2) [inline addons/hr_attendance/views/hr_attendance_overtime_view.xml:80]: same here
```suggestion
    <menuitem id="menu_hr_attendance_overtime_reporting" name="Extra Hours"
        parent="menu_hr_attendance_reporting" sequence="15"
        groups="hr_attendance.group_hr_attendance_officer" action="hr_attendance_overtime_action"/>
```

**Review by Bertrand2 — APPROVED** [review]

**Comment by Bertrand2** [comment]
robodoo r+

**Pull request closed without merging**